fixtures/flight/server/global.js

```diff
@@ -138,10 +144,19 @@ app.all('/', async function (req, res, next) {
       // For HTML, we're a "client" emulator that runs the client code,
       // so we start by consuming the RSC payload. This needs a module
       // map that reverse engineers the client-side path to the SSR path.
-      const root = await createFromNodeStream(rscResponse, moduleMap);
+      let root;
+      let Root = () => {
+        if (root) {
+          return root;
+        }
+        root = createFromNodeStream(rscResponse, ssrBundleConfig.ssrManifest, {
+          moduleLoading: ssrBundleConfig.moduleLoading,
+        });
```

Comment:
Does this suggest that this should just be a single argument option that has both in it?

Reply:
That's what my original implementation had and your feedback at that time was preinitialization and ssr module mapping were unrelated concepts and I should separate them :)

The moduleLoading is technically optional whereas the ssrManifest is a must so I can see the argument they are separate.

Reply:
There's an internal concept and an external concept. This is also related to the name of the type. Almost always we refer to the internal type as one thing but the exported type (which is the name we recommend DefinitelyTyped to use) could be aliased to another name.

I wonder if that's also the case here. Internally they can be two separate concepts and such the configs have to deal with them separately but we can join them here as a single input. Especially if we expect the Webpack plugin to generate it. I don't think I expected that we'd generate it originally.